src/error.rs

```diff
@@ -3,6 +3,7 @@ macro_rules! impl_error {
         $string:literal <=> $variant:ident,
     )*) => {
         /// All [parsing errors](https://html.spec.whatwg.org/#parse-errors) this tokenizer can emit.
+        #[non_exhaustive]
```

Comment:
I'm fine with having people exhaustively match this enum if they need to. I'll just consider it a breaking change to add variants. Since this enum is supposed to match the spec, I think it will not change too often.